src/test/ui/destructuring-assignment/tuple_struct_destructure_fail.stderr

```diff
@@ -15,22 +15,26 @@ LL |     Enum::SingleVariant(a, .., b, ..) = Enum::SingleVariant(0, 1);
    |                            previously used here
 
 error[E0023]: this pattern has 3 fields, but the corresponding tuple struct has 2 fields
-  --> $DIR/tuple_struct_destructure_fail.rs:30:5
+  --> $DIR/tuple_struct_destructure_fail.rs:30:16
    |
 LL | struct TupleStruct<S, T>(S, T);
    | ------------------------------- tuple struct defined here
 ...
 LL |     TupleStruct(a, a, b) = TupleStruct(1, 2);
-   |     ^^^^^^^^^^^^^^^^^^^^ expected 2 fields, found 3
+   |     -----------^^^^^^^^^ expected 2 fields, found 3
+   |     |
+   |     this tuple struct
```

Comment:
This is what I understood from

> with a secondary span pointing at the pattern path.

Is this what you had in mind? I'm wondering if could be confusing because it makes it sound a bit like this is where the tuple struct/variant is defined. Is there a way to add a secondary span *without* a label?

Reply:
> Is this what you had in mind?

Yes, but with one further tweak:

```
LL |     TupleStruct(a, a, b) = TupleStruct(1, 2);
   |     ----------- ^  ^  ^ expected 2 fields, found 3
```

To do that, instead of taking the whole span and removing the beginning, you need to do something like

```rust
let field_spans: MultiSpan = fields.iter().map(|f| f.span).collect::<Vec<_>>().into();
```

> Is there a way to add a secondary span without a label?

`err.span_label(span, "");` 😅

Reply:
It'd be also great if we could also do

```
LL | struct TupleStruct<S, T>(S, T);
   |        -----------       -  - tuple struct has 2 fields
   |        |
   |        tuple struct defined here
```